libwget/iri.c

```diff
@@ -129,6 +130,30 @@ int wget_iri_isunreserved_path(char c)
 	return c > 32 && c < 127 && (c_isalnum(c) || _iri_isunreserved(c) || c == '/');
 }
 
+int wget_iri_is_ipv4_address(const char *host)
+{
+	struct in_addr dst;
+
+	if (!host)
+		return 0;
+
+	return inet_pton(AF_INET, host, &dst);
+}
+
+int wget_iri_is_ipv6_address(const char *host)
+{
+#ifdef HAVE_IPV6
+	struct in6_addr dst;
+
+	if (!host)
+		return 0;
+
+	return inet_pton(AF_INET6, host, &dst);
+#else
+	return 0;
+#endif
+}
+
```

Comment:
These function might come handy at other places too. So IMO, they should not belong to the 'iri' namespace.
I guess, they are best placed in 'net' (e.g. wget_net_is_ipv4_address). If in doubt, you could also make them 'private' functions within iri.c. At a later point, if the need arises, we can make them public.
I am not 100% sure about the latter point, so WDYT ?